test/test_library/unit/test_database.py

```diff
@@ -0,0 +1,971 @@
+        ds1 = DatasetFactory()
+        ds1_vid = ds1.vid
+
+        # testing
+        self.sqlite_db.remove_dataset(ds1.vid)
+        self.assertEquals(
+            self.query(Dataset).filter_by(vid=ds1_vid).all(),
+            [],
+            'Dataset was not removed.')
+
+    # .remove_partition_record tests
+    def test_removes_partition_and_stat(self):
+
+        # prepare state
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+        partition1 = PartitionFactory(dataset=ds1, table=table1)
+        column1 = ColumnFactory(table=table1)
+        colstat1 = ColumnStatFactory(p_vid=partition1.vid, c_vid=column1.vid, d_vid=ds1.vid)
+        self.sqlite_db.session.commit()
+
+        # save necessary ids for later use.
+        partition_vid = partition1.vid
+        colstat1_c_vid = colstat1.c_vid
+
+        # testing
+        self.sqlite_db.remove_partition_record(partition_vid)
+        partition_query = self.query(Partition).filter_by(vid=partition_vid)
+        self.assertEquals(partition_query.all(), [], 'Partition was not deleted.')
+
+        colstat_query = self.query(ColumnStat).filter_by(c_vid=colstat1_c_vid)
+        self.assertEquals(colstat_query.all(), [], 'ColumnStat instance was not deleted.')
+
+    # .get tests
+    # TODO:
+
+    # .get_table tests
+
+    def test_returns_table(self):
+
+        # prepare state
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+
+        # testing
+        ret = self.sqlite_db.get_table(table1.vid)
+        self.assertIsInstance(ret, Table)
+        self.assertEquals(ret.vid, table1.vid)
+
+    # .tables tests
+    @unittest.skip(
+        '.tables() method raises TypeError: list indices must be integers exception and seems unused.')
+    def test_dict_with_all_tables(self):
+
+        # prepare state
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+
+        ds2 = DatasetFactory()
+        table2 = TableFactory(dataset=ds2)
+
+        # testing
+        ret = self.sqlite_db.tables()
+        self.assertIsInstance(ret, dict)
+        self.assertIn(table1.name, ret)
+        self.assertIn(table2.name, ret)
+
+    # .list tests
+    # TODO:
+
+    # .all_vids tests
+    def test_returns_all_datasets_and_partitions(self):
+        # prepare state
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+        self.sqlite_db.session.commit()
+
+        partition1 = PartitionFactory(dataset=ds1, t_id=table1.id_)
+        partition2 = PartitionFactory(dataset=ds1, t_id=table1.id_)
+        self.sqlite_db.session.commit()
+
+        # testing
+        ret = self.sqlite_db.all_vids()
+        self.assertEquals(len(ret), 3)
+        self.assertIn(ds1.vid, ret)
+        self.assertIn(partition1.vid, ret)
+        self.assertIn(partition2.vid, ret)
+
+    # .datasets tests
+    @unittest.skip('raises `AttributeError: type object \'Dataset\' has no attribute \'location\'` error.')
```

Comment:
This line - https://github.com/nmb10/ambry/blob/issue-8/ambry/orm.py#L443 ? But it is not Dataset model field and it does not have table field. So, it is impossible to use sqlalchemy .filter using 'location'. 
```python
>>> self.session.query(Dataset).filter_by(vid=ROOT_CONFIG_NAME_V)                                   
<sqlalchemy.orm.query.Query object at 0xb5ea60ac>
>>> self.session.query(Dataset).filter_by(location=Dataset.LOCATION.LIBRARY)                       
*** InvalidRequestError: Entity '<class 'ambry.orm.Dataset'>' has no property 'location'
```
Am I wrong?